backend/Squirrel.Core/Squirrel.Core.BLL/Interfaces/IMongoService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Squirrel.Core.DAL.Entities;
+
+namespace Squirrel.Core.BLL.Interfaces
+{
+    public interface IMongoService<T> where T : Entity<long>
```

Comment:
why `Entity<long>`?

Reply:
I used Entity<long> to maintain consistency with our AuditEntity<T> pattern. This approach ensures clarity and coherence throughout codebase. Correct me if I'm wrong, please. Thanks!

Reply:
I mean why `long`?

Reply:
As far as I am concerned C# does not support "generics on generics", so picked long as it was used in Sampe entity class. I think this too will be more clear in future when we will have a full picture of MongoDb usage and I will change it, if needed.

Reply:
thanks for the explanation!